src/Gaufrette/Adapter/Ftp.php

```diff
@@ -582,14 +551,17 @@ private function connect()
     /**
      * Closes the adapter's ftp connection.
      */
-    public function close()
+    public function close(): void
     {
         if ($this->isConnected()) {
             ftp_close($this->connection);
         }
     }
 
-    private function isLinuxListing($info)
+    /**
+     * @param array<int, string>|false $info
```

Comment:
```suggestion
     * @param array<int, string>|bool $info
```

Reply:
Event with false... this method is completly fuck-up !